lib/Net/DRI/Protocol/RRI/Domain.pm

```diff
@@ -541,35 +542,31 @@ sub trade
 sub transit {
  my ($rri, $domain, $rd) = @_;
  my $mes = $rri->message();
- my $disconnect = ( exists($rd->{disconnect}) && $rd->{disconnect} eq 'true' ) ? { disconnect => 'true'} : undef;
+ my $disconnect = ( exists($rd->{disconnect}) && $rd->{disconnect} eq 'true' ) ? { disconnect => 'true' } : { disconnect => 'false' };
  my %ns = map { $_ => $mes->ns->{$_}->[0] } qw(domain dnsentry xsi);
  my @d = build_command($mes, 'transit', $domain, $disconnect, \%ns);
 
  $mes->command_body(\@d);
  return;
 }
 
-sub migrate_descr {
+sub create_authinfo1 {
  my ($rri, $domain, $rd) = @_;
  my $mes = $rri->message();
  my %ns = map { $_ => $mes->ns->{$_}->[0] } qw(domain dnsentry xsi);
- my @d = build_command($mes, 'migrate-descr', $domain, undef, \%ns);
-
- ## Contacts, Holder is required
- push @d,build_contact($rd->{contact}) if Net::DRI::Util::has_contact($rd);
-
+ my $hash = exists($rd->{'authinfohash'}) ? { hash => $rd->{'authinfohash'}} : undef;
+ $hash->{'expire'} = $rd->{'authinfoexpire'} if ($hash && exists($rd->{'authinfoexpire'}));
+ my $cmd = ($hash) ? 'createAuthInfo1' : 'createAuthInfo2';
```

Comment:
What was the thinking behind adding a new function etc for createAuthInfo2 when it works via createAuthInfo1? I haven't looked too deep into it, but having 2 authcodes is not new, so I am trying to work out what has changed that has made you split this into two functions?

Reply:
Hum... you are right. Didn't notice _$cmd_ on line 559 that handle 1 and 2 depending of the existence of a hash. My bad!

Restored to how it was and tests updated to reflect the changes:
- https://github.com/thedarkwinter/Net-DRI/pull/343/commits/c1c34ff8e3bde935c265f3f4fe4b490e166e2822